tests/drivers/can/api/src/canfd.c

```diff
@@ -265,6 +293,79 @@ ZTEST(canfd, test_send_receive_mixed)
 		     &test_std_fdf_frame_1, &test_std_frame_2);
 }
 
+/**
+ * @brief Test TX callback is executed when CAN gets stopped.
+ */
+ZTEST(canfd, test_tx_callback_when_can_stops)
+{
+	int err;
+
+	k_sem_reset(&tx_callback_sem);
+
+	send_test_frame_nowait(can_dev, &test_std_fdf_frame_1, tx_std_callback_netdown_1);
+	send_test_frame_nowait(can_dev, &test_std_fdf_frame_2, tx_std_callback_netdown_2);
+
+	err = can_stop(can_dev);
+	zassert_equal(err, 0, "failed to stop CAN controller (err %d)", err);
+
+	err = k_sem_take(&tx_callback_sem, TEST_SEND_TIMEOUT);
+	zassert_equal(err, 0, "missing TX callback");
+
+	err = k_sem_take(&tx_callback_sem, TEST_SEND_TIMEOUT);
+	zassert_equal(err, 0, "missing TX callback");
+
+	err = can_start(can_dev);
+	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
+}
+
+/**
+ * @brief Test error when sending invalid frame (too big data payload).
+ *
+ * This basic test work since the CAN controller is in loopback mode and
+ * therefore ACKs its own frame.
+ */
+ZTEST_USER(canfd, test_send_invalid_frame)
+{
+	struct can_frame test_invalid_frame = {
+		.flags   = CAN_FRAME_FDF | CAN_FRAME_BRS,
+		.flags   = 0,
+		.id      = TEST_CAN_STD_ID_1,
+		.dlc     = 9,
+		.data    = {0}
+	};
+	int err;
+
+	err = can_send(can_dev, (const struct can_frame *) &test_invalid_frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_not_equal(err, -EBUSY, "arbitration lost in loopback mode");
+	zassert_equal(err, -EINVAL, "send incorrect frame (err %d)", err);
+
+	test_invalid_frame.flags = CAN_FRAME_IDE | CAN_FRAME_FDF | CAN_FRAME_BRS;
+	test_invalid_frame.dlc = 65;
+
+	err = can_send(can_dev, (const struct can_frame *) &test_invalid_frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_not_equal(err, -EBUSY, "arbitration lost in loopback mode");
+	zassert_equal(err, -EINVAL, "send incorrect frame (err %d)", err);
+}
```

Comment:
Please have a look at how this is done in `classic.c` using `send_invalid_frame()`. This needs to take `CONFIG_RUNTIME_ERROR_CHECKS` into account.

Reply:
I added `Z_TEST_SKIP_IFNDEF(CONFIG_RUNTIME_ERROR_CHECKS);`